Celigo.ServiceManager.NetSuite.REST/src/ServiceCollection.extensions.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Xml.Schema;
+using Celigo.ServiceManager.NetSuite.REST;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    using Options = Microsoft.Extensions.Options.Options;
+    
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddNetSuiteRestClientSupport(this IServiceCollection services, IConfiguration configuration)
+        {
+            services
+                .AddOptions()
+                .Configure<RestClientOptions>(configuration.GetSection(RestClientOptions.ConfigurationSectionName))
+                .AddHttpClient()
+                ;
+
+            return services;
+        }
+
+        public static IServiceCollection AddRestletClient(this IServiceCollection services,
+                                                            IConfiguration configuration, 
+                                                            params RestletConfig[] restlets)
+        {
+            services
+                .AddNetSuiteRestClientSupport(configuration)
+                .Configure<RestletConfig>(configuration.GetSection(RestletConfig.ConfigurationSectionName))
+                .Configure<RestletConfigOptions>(configuration.GetSection(RestletConfig.ConfigurationSectionName));
+
+            if (restlets.Length > 0)
+            {
+                var restletConfigOptions = new RestletConfigOptions { Restlets = restlets };
+                if (restlets.Any(r => string.IsNullOrEmpty(r.RestletName)))
+                {
+                    throw new ArgumentNullException($"{nameof(RestletConfig)}.{nameof(RestletConfig.RestletName)}");
+                }
+                services.AddSingleton(Options.Create(restletConfigOptions));
+            }
+
+            services.AddHttpClient<IRestletClient, RestletClient>();

```

Comment:
That's not how I think it works 🤔 

When we call `IHttpClientFactory.CreateClient(name)` it's looking for a cached `HttpClient` instance. If not, it creates and caches one. Doing the above iterative `AddHttpClient` probably helps pre-caching it.
Also, since we are only using `IHttpClientFactory` in `RestletClientFactory` to create `HttpClient`s rather than `RestletClient`s, above wouldn't matter anyway, right? Or did I miss something here?

Reply:
Ok, I thought that won't work. 

Actually [CreateClient](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.ihttpclientfactory.createclient?view=dotnet-plat-ext-3.1#remarks)  will always return a new client. But  underlying [HttpMessageHandler](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpmessagehandler?view=dotnet-plat-ext-3.1) seems to be cached and reused by factory.

Reply:
Correct.